### PR TITLE
Feature zap-api-scan with options

### PR DESCRIPTION
**Pull request by nguyennhatthuan**
Implement feature zap-api-scan
Base on idea of Zap action full scan
@psiinon please help me to review the PR

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+const core = require("@actions/core");
+const exec = require("@actions/exec");
+const common = require("@zaproxy/actions-common-scans");
+const _ = require("lodash");
+
+// Default file names
+let mdReportName = "report_md.md";
+let htmlReportName = "report_html.html";
```
Comment (psiinon) [inline index.js]: No jsonReportName? The other 2 actions support this

**Review by psiinon — CHANGES_REQUESTED** [review]
Just spotted one issue, the missing json report option

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+inputs:
+  token:
+    description: "GitHub Token to create issues in the repository"
+    required: false
+    default: ${{ github.token }}
+  target:
+    description: "Target URL"
+    required: true
+  format:
+    description: "Format openapi or soap"
```
Comment (lghakamo-paf) [inline action.yml]: Inconsistent with README file which states `openapi, soap, or graphql`

Comment (kingthorin) [reply action.yml]: Good catch

**Comment by psiinon** [comment]
@nguyennhatthuan are you able to address these comments?

**Comment by nguyennhatthuan** [comment]
hi!

Sorry that I was a bit busy last week. I'll have a look and work on the
comments this weekend! :D

With sincere thanks and really appreciate you're spending your valuable
time reading it!
---------------------------------------------
*Thuan Nguyen *
*Mobile*: (+84) 398 681 685
*Email*: ***@***.***
L ***@***.***>inkedIn:
https://www.linkedin.com/in/thuannguyennhat
*Skype*: thuannguyennhat


Vào Th 5, 12 thg 8, 2021 vào lúc 16:19 Simon Bennetts <
***@***.***> đã viết:

**Comment by psiinon** [comment]
No problem and thanks for doing this!

File: index.js (hunk 1 above)
Comment (nguyennhatthuan) [reply index.js]: hi @psiinon, I've updated the code with support jsonReportName like the other 2 actions 😄

File: action.yml (hunk 2 above)
Comment (nguyennhatthuan) [reply action.yml]: thanks @Ighakamo @kingthorin! I've updated the description. I based on this article 😄 
https://www.zaproxy.org/docs/docker/api-scan/

**Comment by nguyennhatthuan** [comment]
hi @psiinon ! I've udpated the PR, please help me to review it again 😄 My apologies for the late response on this.

**Review by psiinon — APPROVED** [review]

**Comment by psiinon** [comment]
Looks good to me, sorry for taking so long to review it :/

**Review by lghakamo-paf — APPROVED** [review]
LGTM

**Review by psiinon — CHANGES_REQUESTED** [review]
Oh, the dist/index.js file should be removed - that will be generated when we release the action.

**Review by psiinon — APPROVED** [review]
But we can always regenerate it anyway, so no need to remove :)

**Comment by thc202** [comment]
Worth adding zaproxy/action-baseline#56.

**Comment by nguyennhatthuan** [comment]
hi @thc202, @psiinon, so I would need to follow the PR zaproxy/action-baseline#56 for adding allowIssueWriting option or it'll be handled by your team? :)

**Comment by thc202** [comment]
We can do that.

**Comment by nguyennhatthuan** [comment]
hi @psiinon @thc202 , so now can we merge and then update to add fix issue writing later or have to wait for the fixing added? 
Will need approval from @thc202 😄

**Comment by thc202** [comment]
This is being worked on, expect changes shortly.

**Comment by nguyennhatthuan** [comment]
Thanks for your update! 😄

**Comment by thc202** [comment]
@nguyennhatthuan could you take a look too? (The indentation changes messed the diff, just added the new input though.)

I didn't regenerate the dist yet, will do once it's OK'ed (and also fixup).

**Comment by thc202** [comment]
@sshniro ping (in case you are not yet watching this repo).

File: index.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        let repoName = process.env.GITHUB_REPOSITORY;
+        let token = core.getInput('token');
+        let docker_name = core.getInput('docker_name');
+        let target = core.getInput('target');
+        let format = core.getInput('format');
+        let rulesFileLocation = core.getInput('rules_file_name');
+        let cmdOptions = core.getInput('cmd_options');
+        let issueTitle = core.getInput('issue_title');
+        let failAction = core.getInput('fail_action');
+        let allowIssueWriting = core.getInput('allow_issue_writing');
```
Comment (sshniro) [inline index.js:25]: If you are using action common version `0.2.0`, then please do convert this to [boolean](https://github.com/zaproxy/action-baseline/pull/59/commits/a9dcad2ca65f739db41f88210c69cceabb2a4c34)

**Review by sshniro — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (thc202) [reply index.js:25]: Good catch, I thought it was already doing that. Will push that change.

Comment (thc202) [reply index.js:25]: Done.

**Review by sshniro — APPROVED** [review]

**Review by kingthorin — APPROVED** [review]

**Comment by thc202** [comment]
Regenerated the dist and fixed up the commits.

**Review by thc202 — APPROVED** [review]

**Comment by psiinon** [comment]
@nguyennhatthuan FYI it has now been published! https://github.com/marketplace/actions/owasp-zap-api-scan
Thanks for doing this.
How would you like to be included in the ZAP credits? Do you have a twitter account or a site you would like us to link to?
https://www.zaproxy.org/docs/desktop/credits/

**Comment by nguyennhatthuan** [comment]
Thanks for your inform @psiinon
I'll apply to my projects 😍

With sincere thanks and really appreciate you're spending your valuable
time reading it!
---------------------------------------------
*Thuan Nguyen *
*Mobile*: (+84) 398 681 685
*Skype*: thuannguyennhat


Vào Th 3, 21 thg 9, 2021 vào lúc 20:45 Simon Bennetts <
***@***.***> đã viết:

**Pull request merged**